render/hyperscript.js

```diff
@@ -78,6 +78,11 @@ function hyperscript(selector) {
 	} else if (typeof attrs !== "object" || attrs.tag != null || Array.isArray(attrs)) {
 		attrs = {}
 		start = 1
+	} else {
+		attrs = Object.keys(attrs).reduce(function(newAttributes, key) {
+			newAttributes[key] = attrs[key]
+			return newAttributes
+		}, {})
```

Comment:
Just use a (`hasOwnProperty` guarded) `for ... in` loop. Keep in mind this is in a somewhat performance-critical place, and we'd rather *not* be creating a closure here.

Also, this copying should only be done in `execSelector`, preferably only if `state.attrs` there is non-empty. That way, it is only done when it's absolutely necessary, not on every call.

Keep in mind, each render could literally create thousands of these nodes each time. It has to be fast in the common path.